src/AutoRest.CSharp/Common/Output/Models/Types/SystemObjectType.cs

```diff
@@ -0,0 +1,174 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using AutoRest.CSharp.Generation.Types;
+using AutoRest.CSharp.Input;
+using AutoRest.CSharp.Output.Models.Requests;
+using AutoRest.CSharp.Output.Models.Serialization;
+using AutoRest.CSharp.Output.Models.Shared;
+
+namespace AutoRest.CSharp.Output.Models.Types
+{
+    internal class SystemObjectType : ObjectType
+    {
+        private Type _type;
+        private CSharpType _csharpType;
+
+        public SystemObjectType(Type type, BuildContext context)
+            : base(context)
+        {
+            _type = type;
+            _csharpType = new CSharpType(type);
+            SystemType = type;
+        }
+
+        public Type SystemType { get; }
+
+        public override ObjectTypeProperty? AdditionalPropertiesProperty => null;
+
+        public override bool IncludeSerializer => false;
+
+        public override bool IncludeDeserializer => false;
+
+        public override bool IncludeConverter => false;
+
+        public override string DefaultName => GetNameWithoutGeneric(_type);
+
+        private string ToCamelCase(string name)
+        {
+            return name.Substring(0, 1).ToLower(CultureInfo.InvariantCulture) + name.Substring(1);
+        }
+
+        private IEnumerable<PropertyInfo> _myProperties => _type.GetProperties().Where(p => p.DeclaringType == _type);
```

Comment:
Inline?

Reply:
sure, i was originally using it in multiple places